From 4775bb34c485c1efed2c20d72faba5adf97147d3 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tor=20Arne=20Vestb=C3=B8?= <tor.arne.vestbo@qt.io>
Date: Mon, 6 Oct 2025 16:08:10 +0200
Subject: [PATCH 306/553] macOS: Use SF Symbols chevrons for toolbar extension
 icons

To match the native look of NSToolbar.

Fixes: QTBUG-140689
Pick-to: 6.8
Change-Id: Ib21d7e490d36ecb62d6267306661a4ab02063ed6
Reviewed-by: Volker Hilsheimer <volker.hilsheimer@qt.io>
(cherry picked from commit 379b354c15de4c6bbb8f5e476ea4deecb11827cd)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/plugins/platforms/cocoa/qcocoatheme.mm |  4 ++++
 src/plugins/styles/mac/qmacstyle_mac.mm    | 24 ----------------------
 src/plugins/styles/mac/qmacstyle_mac_p.h   |  2 --
 src/widgets/styles/qcommonstyle.cpp        |  2 ++
 4 files changed, 6 insertions(+), 26 deletions(-)

diff --git a/src/plugins/platforms/cocoa/qcocoatheme.mm b/src/plugins/platforms/cocoa/qcocoatheme.mm
index 12a1bc2fd00..b784a7d362e 100644
--- a/src/plugins/platforms/cocoa/qcocoatheme.mm
+++ b/src/plugins/platforms/cocoa/qcocoatheme.mm
@@ -387,6 +387,10 @@ QPixmap QCocoaTheme::standardPixmap(StandardPixmap sp, const QSizeF &size) const
     case FileIcon:
         iconType = kGenericDocumentIcon;
         break;
+    case ToolBarHorizontalExtensionButton:
+        return QIcon::fromTheme("chevron.forward.2").pixmap(size.toSize());
+    case ToolBarVerticalExtensionButton:
+        return QIcon::fromTheme("chevron.down.2").pixmap(size.toSize());
     default:
         break;
     }
diff --git a/src/plugins/styles/mac/qmacstyle_mac.mm b/src/plugins/styles/mac/qmacstyle_mac.mm
index de8ebbeebf5..7aa7f03c682 100644
--- a/src/plugins/styles/mac/qmacstyle_mac.mm
+++ b/src/plugins/styles/mac/qmacstyle_mac.mm
@@ -6955,30 +6955,6 @@ bool QMacStyle::event(QEvent *e)
     return false;
 }
 
-QIcon QMacStyle::standardIcon(StandardPixmap standardIcon, const QStyleOption *opt,
-                              const QWidget *widget) const
-{
-    switch (standardIcon) {
-    default:
-        return QCommonStyle::standardIcon(standardIcon, opt, widget);
-    case SP_ToolBarHorizontalExtensionButton:
-    case SP_ToolBarVerticalExtensionButton: {
-        QPixmap pixmap(QLatin1String(":/qt-project.org/styles/macstyle/images/toolbar-ext-macstyle.png"));
-        if (standardIcon == SP_ToolBarVerticalExtensionButton) {
-            QPixmap pix2(pixmap.height(), pixmap.width());
-            pix2.setDevicePixelRatio(pixmap.devicePixelRatio());
-            pix2.fill(Qt::transparent);
-            QPainter p(&pix2);
-            p.translate(pix2.width(), 0);
-            p.rotate(90);
-            p.drawPixmap(0, 0, pixmap);
-            return pix2;
-        }
-        return pixmap;
-    }
-    }
-}
-
 int QMacStyle::layoutSpacing(QSizePolicy::ControlType control1,
                              QSizePolicy::ControlType control2,
                              Qt::Orientation orientation,
diff --git a/src/plugins/styles/mac/qmacstyle_mac_p.h b/src/plugins/styles/mac/qmacstyle_mac_p.h
index 6a8c9576e8f..078e8894a3e 100644
--- a/src/plugins/styles/mac/qmacstyle_mac_p.h
+++ b/src/plugins/styles/mac/qmacstyle_mac_p.h
@@ -76,8 +76,6 @@ public:
 
     bool event(QEvent *e);
 
-    QIcon standardIcon(StandardPixmap standardIcon, const QStyleOption *opt = nullptr,
-                       const QWidget *widget = nullptr) const;
     int layoutSpacing(QSizePolicy::ControlType control1, QSizePolicy::ControlType control2,
                       Qt::Orientation orientation, const QStyleOption *option = nullptr,
                       const QWidget *widget = nullptr) const;
diff --git a/src/widgets/styles/qcommonstyle.cpp b/src/widgets/styles/qcommonstyle.cpp
index d2ac4a48666..129fd05d3d5 100644
--- a/src/widgets/styles/qcommonstyle.cpp
+++ b/src/widgets/styles/qcommonstyle.cpp
@@ -5834,6 +5834,8 @@ QIcon QCommonStylePrivate::iconFromMacTheme(QCommonStyle::StandardPixmap standar
         case QStyle::SP_DirLinkIcon:
         case QStyle::SP_FileLinkIcon:
         case QStyle::SP_FileIcon:
+        case QStyle::SP_ToolBarHorizontalExtensionButton:
+        case QStyle::SP_ToolBarVerticalExtensionButton:
             if (const QPlatformTheme *theme = QGuiApplicationPrivate::platformTheme()) {
                 QPlatformTheme::StandardPixmap sp = static_cast<QPlatformTheme::StandardPixmap>(standardIcon);
                 QIcon retIcon;
-- 
2.51.2

